app/Models/Contest.php

```diff
@@ -57,9 +59,23 @@ public function getAttachmentsAttribute()
         return $this->getMedia('contest_attachments');
     }
 
+    public function getStateAttribute()
+    {
+        if (is_null($this->announced_at) || $this->announced_at->isFuture()) {
+            return 'upcoming';
+        } elseif (
+            ($this->announced_at->isPast()) &&
+            (is_null($this->finished_at) || $this->finished_at->isFuture())
+        ) {
+            return 'ongoing';
+        }
```

Comment:
```suggestion
        } 
        if (is_null($this->finished_at) || $this->finished_at->isFuture()) {
            return 'ongoing';
        }
```

I think `$this->announced_at->isPast()` is implied from the first `if`.